### DISCUSSION TITLE
Consider adjustments to sizes to default to whole numbers

### DISCUSSION THREAD

**Discussion by jeffwllms** [argyleink/open-props#195] [Ideas]
Should the sizing open-props provides default to whole numbers.   While I think for the most part the do, but there are a couple places where they don't.

For example in Typography.

There is a "font-size" of 1.1rem.  at the default 16px this calcs to 17.6px.  A font size of 1.125 would give an 18.  

I believe this exists on line height as well.  there is a 1.1 but that isn't rem and is relative to the actual font size of the element itself.  So maybe not as much of an issue. 

Anyway, maybe I am a bit old-school and sub-pixels aren't much a thing to care about anymore.  But if it is something that we wanted to look at I would be happy to see if there are any other places it comes up. Although, any changes would effect peoples currently using the sizes, that is likely a bigger problem.

**Comment by jeffwllms** [comment]
Maybe this only exists in the font sizing as well, which also might not be a big deal since fonts aren't really in whole pixels themselves.

**Comment by argyleink** [comment]
i could see wanting to avoid sub pixel sizes, makes sense! consider just overwriting them individually with your own exceptions?

```css
/* don't have to specificity fight, just set a new value */
:root {
  --font-size-2: 1.125rem;
}
```

**Comment by jeffwllms** [comment]
> i could see wanting to avoid sub pixel sizes, makes sense! consider just overwriting them individually with your own exceptions?
> 
> ```css
> /* don't have to specificity fight, just set a new value */
> :root {
>   --font-size-2: 1.125rem;
> }
> ```

This is the route I am taking now, just was wondering if this was worth looking into on the defaults. ;)  but I suppose making changes like that to the defaults wouldn't be ideal.  I will search for other ways to contribute, lol.

**Comment by argyleink** [comment]
i'd strongly consider a PR btw 🙂  

i mean, it's a small change, may go unnoticeable to devs and users alike, but perhaps the engine will appreciate it. it sounds more deliberate than what i've done here, which is nice! are [there others than could benefit](https://github.com/argyleink/open-props/blob/main/src/props.fonts.js#L45-L51) from a similar adjustment?

**Comment by WebMechanic** [comment]
With regards to the font-sizes: as soon as you pick some other (web) font, those defaults don't work anymore anyways and need some tweaking. I personaly use a scale based on a mix of Fibunacci numbers and Golden Ratio "rounded" to .25 to make math easier. I also apply them to white space.

Just like colours, the font scale's a matter of taste and depends on the layout and design anyways. They're a good starting point, tho.

It's been one of the first files I copied and virtually replaced the entirety of values in it and also adjusted the fluid values (which I like btw) to use the font-size variables as well and be less exaggregated - depends on the font.
